src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -367,9 +370,37 @@ public static string UrlEncode(string value)
             if (string.IsNullOrEmpty(value))
                 return value;
 
-            byte[] bytes = Encoding.UTF8.GetBytes(value);
-            byte[] encodedBytes = UrlEncode(bytes, 0, bytes.Length, false /* alwaysCreateNewReturnValue */);
-            return Encoding.UTF8.GetString(encodedBytes, 0, encodedBytes.Length);
+            int cSafe = 0;
+            int cSpaces = 0;
+            for (int i = 0; i < value.Length; i++)
+            {
+                char ch = value[i];
+                if (ch == ' ')
+                {
+                    cSpaces++;
+                }
+                else if (IsUrlSafeChar(ch))
+                {
+                    cSafe++;
+                }
```

Comment:
Minor, but we probably expect url safe chars to be more common than spaces, right?  I'm wondering if the order of these checks should be reversed.  Doing so would match the order done earlier, and would mean that for url safe chars, we don't have to first check for whether it's a space.